trustfall_core/src/ir/types.rs

```diff
@@ -95,30 +93,13 @@ impl NamedTypedValue for Argument {
     }
 }
 
-pub(crate) fn are_base_types_equal_ignoring_nullability(left: &BaseType, right: &BaseType) -> bool {
-    match (left, right) {
-        (BaseType::Named(l), BaseType::Named(r)) => l == r,
-        (BaseType::List(l), BaseType::List(r)) => {
-            are_base_types_equal_ignoring_nullability(&l.base, &r.base)
-        }
-        (BaseType::Named(_), BaseType::List(_)) | (BaseType::List(_), BaseType::Named(_)) => false,
-    }
-}
-
-pub(crate) fn is_base_type_orderable(operand_type: &BaseType) -> bool {
-    match operand_type {
-        BaseType::Named(name) => {
-            name == "Int" || name == "Float" || name == "String" || name == "DateTime"
-        }
-        BaseType::List(l) => is_base_type_orderable(&l.base),
-    }
+pub(crate) fn are_base_types_equal_ignoring_nullability(left: &Type, right: &Type) -> bool {
+    left.base_named_type() == right.base_named_type()
```

Comment:
This is not a correct implementation. It does not match the semantics of the original function. Please write tests to catch the bug then update the implementation.

Please also review your other implementation changes in this module for similar errors.